model/flow/identity.go

```diff
@@ -508,3 +562,59 @@ func (il IdentityList) EqualTo(other IdentityList) bool {
 	}
 	return true
 }
+
+// Exists takes a previously sorted Identity list and searches it for the target value
+// This code is optimized, so the coding style will be different
+// target:  value to search for
+// CAUTION:  The identity list MUST be sorted prior to calling this method
+func (il IdentityList) Exists(target *Identity) bool {
+	return il.IdentifierExists(target.NodeID)
+}
+
+// Exists takes a previously sorted Identity list and searches it for the target value
+// target:  value to search for
+// CAUTION:  The identity list MUST be sorted prior to calling this method
+func (il IdentityList) IdentifierExists(target Identifier) bool {
+	left := 0
+	lenList := len(il)
+	right := lenList - 1
+	mid := int(uint(right) >> 1)
+	num2 := target[:]
+
+	// pre-calculate these 4 values for comparisons later
+	var tgt [4]uint64
+	tgt[0] = binary.BigEndian.Uint64(num2[:])
+	tgt[1] = binary.BigEndian.Uint64(num2[8:])
+	tgt[2] = binary.BigEndian.Uint64(num2[16:])
+	tgt[3] = binary.BigEndian.Uint64(num2[24:])
+
+	for {
+		num1 := il[mid].NodeID[:]
+		lenID := len(num1)
+		i := 0
+
+		for {
+			chunk1 := binary.BigEndian.Uint64(num1[i:])
+			chunk2 := tgt[i/8]
+
+			if chunk1 < chunk2 {
+				left = mid + 1
+				break
+			} else if chunk1 > chunk2 {
+				right = mid - 1
+				break
+			} else if i >= lenID-8 {
+				// we're on the last chunk of 8 bytes, and
+				// so return true if equal -- it exists
+				return true
+			}
+
+			// these 8 bytes were equal, so increment index by 8 bytes
+			i += 8
+		}
+		if left > right {
+			return false
+		}
+		mid = int(uint(left+right) >> 1)
```

Comment:
Do we really need the conversion `int` -> `uint` -> `int` (?) My understanding that the type conversion is not needed for non-negative `int` values.  `left` and `right` are always non-negative. I think a comment here might be sufficient.

same for https://github.com/onflow/flow-go/blob/9120a897c1936819ee321431377d5e0dc4fba3a4/model/flow/identity.go#L581 (provided you include the short cut for zero-length `IdentityList`)

Reply:
will do

Reply:
done